src/streaming/controllers/BufferController.js

```diff
@@ -559,6 +559,24 @@ function BufferController(config) {
             if (currentTimeRequest) {
                 rangeStart = Math.max(currentTimeRequest.startTime + currentTimeRequest.duration, rangeStart);
             }
+
+            // Never remove the contiguous range of targetTime in order to avoid flushes & reenqueues in WebKit based browsers
+            const ua = navigator.userAgent.toLowerCase();
+            if (/webkit/.test(ua) && !/chrome/.test(ua) && !/windows phone/.test(ua)) {
```

Comment:
Please use our UAParser `Utils.parseUserAgent` in order to check for the device.